### PR TITLE
863 date picker docs

### PR DESCRIPTION
**Pull request by gd2910**
## Summary of the changes

Guidance, code and a11y pages for canary date-picker

## Related issue

#863 

## Checklist

- [x] I have raised this pull request against the [develop branch](https://github.com/mi6/ic-design-system/tree/develop)

### CODE REVIEW

File: src/content/structured/components/canary/date-picker/guidance.mdx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+navPriority: 3
+
+date: "2024-04-03"
+
+title: "Date Picker"
+
+status: "CANARY"
+
+subTitle: "Guidance"
```
Comment (GCHQ-Developer-539) [inline src/content/structured/components/canary/date-picker/guidance.mdx:12]:
```suggestion
subTitle: "A date picker is a type of drop-down control that allows people to easily view and select dates."
```

File: src/content/structured/components/canary/date-picker/guidance.mdx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    { title: "Code", path: "/components/canary/date-picker/code" },
+    { title: "Accessibility", path: "/components/date-picker/accessibility" },
+  ]
+---
+
+import { IcDatePicker } from "@ukic/canary-react";
+
+## Introduction
+
+A date picker is a type of drop-down control that allows people to easily view and select dates.
```
Comment (GCHQ-Developer-539) [inline src/content/structured/components/canary/date-picker/guidance.mdx:26]:
```suggestion
An example of a date picker component.
```

File: src/components/ComponentDetails/index.tsx [hunk 3]
```diff
@@ -1,16 +1,23 @@
...
+  let componentDetails = componentJson.components.find(
     (currentComponent) => currentComponent.tag === component
   )!;
 
+  // If the component is not found in the main component JSON, check the canary component JSON
+  if (componentDetails === undefined) {
+    componentDetails = canaryComponentJson.components.find(
+      (currentComponent) => currentComponent.tag === component
+    )!;
+  }
```
Comment (ad9242) [inline src/components/ComponentDetails/index.tsx:20]: just a thought. could it pass a boolean `canary` prop to the ComponentDetails & then this code would be much simpler eg
```
const ComponentDetails: React.FC<{ component: string, canary=false }> = ({ component }) => {
  const jsonData = canary ? componentJson : canaryComponentJson;
  const componentDetails = jsonData.components.find(
  ...
```

**Review by ad9242 — CHANGES_REQUESTED** [review]

File: src/components/ComponentDetails/index.tsx (hunk 3 above)
Comment (GCHQ-Developer-741) [reply src/components/ComponentDetails/index.tsx]: Amended using this suggestion

**Review by GCHQ-Developer-741 — APPROVED** [review]

File: src/content/structured/components/canary/date-picker/guidance.mdx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+### Width
+
+Use the resizing options to increase or decrease width to match other elements in a form.
+
+Use the [date input](/components/canary/date-input) resizing options to change size of the input and calendar between minimum and maximum widths. For example, to create a larger date picker, resize width of [date input](/components/canary/date-input).
+
+<DoDontCaution
+  imageSrc={figure3}
+  imageAlt="Two date picker examples at their maximum and minimum width"
+  caption="Change the width of the [date input] (../date-input/guidance.mdx) to adjust the date picker width."
```
Comment (ad9242) [inline src/content/structured/components/canary/date-picker/guidance.mdx:81]: this is displaying incorrectly: 
image

File: src/content/structured/components/canary/date-picker/guidance.mdx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+  imageSrc={figure4}
+  imageAlt="A date picker field with label, helper and placeholder text highlighted."
+  caption="Labels, helper text and placeholder text make up the instructional elements of an input."
+/>
+
+Refer to [typography styling](/styles/typography#text) for correct label, error text and helper text styles.
+
+### Labels
+
+Use the `start-of-week` prop to alternate between week beginning Monday or Sunday.
```
Comment (ad9242) [inline src/content/structured/components/canary/date-picker/guidance.mdx:98]: this description doesn't match the heading. "labels" - it talks about the start of week prop

**Pull request merged**